File: java/SwapTwoNumbers.java
import java.util.Scanner;

public class SwapTwoNumbers {
    public static void main(String[] args){
        Scanner sc = new Scanner(System.in);

        System.out.print("Enter first number: ");
        int num1 = sc.nextInt();

        System.out.print("Enter second number: ");
        int num2 = sc.nextInt();

        num1 = num1 ^ num2;
        num2 = num1 ^ num2;
        num1 = num1 ^ num2;

        System.out.println("The first number after swapping : " + num1);
        System.out.println("The second number after swapping : " + num2);
        
        sc.close();
    }
}

File: C/swap_two_numbers.c
#include <stdio.h>
int main(){
    int num1 ,num2;

    printf("Enter first number: ");
    scanf("%d",&num1);

    printf("Enter second number: ");
    scanf("%d",&num2);

    printf("---------------------------------\n");
    printf("The first number is : %d\n", num1);
    printf("The second number is : %d\n", num2);
    printf("---------------------------------\n");

    num1 = num1 ^ num2;
    num2 = num1 ^ num2;
    num1 = num1 ^ num2;

    printf("The first number after swapping : %d\n", num1);
    printf("The second number after swapping : %d\n", num2);

    return 0;
}